src/current/v23.2/log-sql-statistics-to-datadog.md

```diff
@@ -82,12 +89,33 @@ SET CLUSTER SETTING sql.telemetry.query_sampling.max_event_frequency = 100000;
 The `sql.telemetry.query_sampling.max_event_frequency` cluster setting and the `buffering` options in the `logs.yaml` control how many events are emitted to Datadog and that can be potentially dropped. Adjust this setting and these options according to your workload, depending on the size of events and the queries per second (QPS) observed through monitoring.
 {{site.data.alerts.end}}
 
-## Step 4. Monitor TELEMETRY logs in Datadog
+## Step 4. Configure CockroachDB to emit query and transaction events (optional)
+
+Enable the [`sql.telemetry.query_sampling.enabled` cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}#setting-sql-telemetry-query-sampling-enabled) so that executed queries and transactions will emit an event on the telemetry [logging channel]({% link {{ page.version.version }}/logging-overview.md %}#logging-channels):
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET CLUSTER SETTING sql.telemetry.query_sampling.enabled = true;
+~~~
+
+Set the [`sql.telemetry.query_sampling.mode` cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}) to `transaction` so that `sampled_query` and `sampled_transaction` events are emitted:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET CLUSTER SETTING sql.telemetry.query_sampling.mode = 'transaction';
+~~~
+
+Configure the following [cluster settings]({% link {{ page.version.version }}/cluster-settings.md %}) to values that are dependent on the level of granularity you require and how much performance impact from frequent logging you can tolerate:
+
+- `sql.telemetry.transaction_sampling.max_event_frequency` (default `8`) is the max event frequency (events per second) at which we sample transactions for telemetry. If sampling mode is set to 'statement', this setting is ignored. In practice, this means that we only sample a transaction if 1/max_event_frequency seconds have elapsed since the last transaction was sampled. 
+- `sql.telemetry.transaction_sampling.statement_events_per_transaction.max` (default `50`) is the maximum number of statement events to log for every sampled transaction. Note that statements that are logged by force do not adhere to this limit.
```

Comment:
> Note that statements that are logged by force do not adhere to this limit. 

What does this mean?

Reply:
How does one force a log?

Reply:
Hi @xinhaoz, would you be able to explain this line? I copied it from the description generated from [code](https://github.com/cockroachdb/cockroach/blame/b777cef4b498369226f2e85fe265ba09a267001a/pkg/sql/telemetry_logging.go#L63):
```

Reply:
Currently we force log on the following conditions:
* if the stmt type is not DML <- that's been around since the inception of statement logging to telemetry AFAIK. This change excludes BEGIN and COMMIT from being force logged despite them being non-dml statements.
* if tracing is enabled
* if `sql.telemetry.query_sampling.internal_console.enabled` is true and the query is from db console (starts with $ internal-console

Reply:
Thanks Xin. Can you specify which tracing? The one governed by `sql.telemetry.query_sampling.enabled`?

Floreceny, Perhaps 'force a log' is the wrong term here. I'd just say that logs are always captured under those conditions (we dont need to call out the db console one since that's only internal)

Reply:
@xinhaoz confirmed that by "tracing is enabled", she was referring to you `SET TRACING=on;`  as documented [here](https://www.cockroachlabs.com/docs/v23.2/set-vars#set-tracing).

I think we need to mention DB Console statements because they spam the telemetry log if the DB Console is used.